src/pages/events.tsx

```diff
@@ -59,13 +65,24 @@ const filterEvent = (
 };
 
 const ROWS_PER_PAGE = 25;
-const EventsPage = ({ events, attendances }: EventsPageProps) => {
+const EventsPage = ({ events, attendances, initialFilters }: EventsPageProps) => {
   const [page, setPage] = useState(0);
-  const [communityFilter, setCommunityFilter] = useState('all');
-  const [dateFilter, setDateFilter] = useState('upcoming');
-  const [attendedFilter, setAttendedFilter] = useState('all');
+  const [communityFilter, setCommunityFilter] = useState(initialFilters.community);
+  const [dateFilter, setDateFilter] = useState(initialFilters.date);
+  const [attendedFilter, setAttendedFilter] = useState(initialFilters.attended);
   const [query, setQuery] = useState('');
 
+  const router = useRouter();
+
+  useEffect(() => {
+    const validState =
+      initialFilters.community === communityFilter &&
+      initialFilters.date === dateFilter &&
+      initialFilters.attended === attendedFilter;
+    if (!validState) router.reload();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router, initialFilters]);
```

Comment:
I think you should add a comment explaining why

Reply:
Also, instead of reloading the page, would it work to just call setCommunityFilter etc on the new initialFilters?

Reply:
if that's the case, you probably don't even need to disable the eslint rule; you can just override the states without checking, and React will rerender only if necessary